app/commands/mentor/discussion/retrieve.rb

```diff
@@ -17,10 +17,7 @@ def initialize(mentor,
                  sorted: true, paginated: true,
                  exclude_uuid: nil)
 
-    # This will be a code-level exception rather than a user-level
-    # exception so we don't worry about a special class. Getting this
-    # tells us that we've got a bug.
```

Comment:
As this is directly callable from the API, I though that it might make sense to have a custom error